### ISSUE TITLE
CreateSondbox account api works from playground but does not work when called from external script after authentication

### ISSUE THREAD

**Issue by jayantsarda** [fastlane/fastlane#20126]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
script:
```
token = Spaceship::ConnectAPI::Token.create(
  key_id: '<key>',
  issuer_id: '<Issuer_id>',
  filepath:  File.absolute_path("./AuthKey_<key>p8")
)

Spaceship::ConnectAPI.token = token

# blow printing app names works ... so its authenticated
Spaceship::ConnectAPI::App.all.collect do |app|
 puts  app.name
end 

# below code does not work from script
testers = Spaceship::ConnectAPI::SandboxTester.create(
  first_name: "Test", # required
  last_name: "Three", # required
  email: "sandbox@test.com", # required
  password: "Passwordtest1", # required. Must contain >=8 characters, >=1 uppercase, >=1 lowercase, >=1 numeric.
  confirm_password: "Passwordtest1", # required
  secret_question: "Question", # required. Must contain >=6 characters
  secret_answer: "Answer", # required. Must contain >=6 characters
  birth_date: "1980-03-01", # required
  app_store_territory: "USA" # required
)
```
I have tried below code from spaceship playground and it works but does not work from script

```
attributes = {}
# Then go through all attributes like this
attributes["firstName"] = "ABC"
attributes["lastName"] = "XYA"
attributes["email"] = "test@gmail.com"
attributes["password"] = "pwd"
attributes["confirmPassword"] = "pwd"
attributes["secretQuestion"] = "Question"
attributes["secretAnswer"] = "Answer"
attributes["birthDate"] = "1980-03-01"
attributes["appStoreTerritory"] = "USA"
Spaceship::ConnectAPI.post_sandbox_tester(attributes: attributes)
```
it throws below error
```
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/api_client.rb:199:in `handle_response': The URL path is not valid - The resource 'sandboxTesters' does not exist (Spaceship::UnexpectedResponse)
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/api_client.rb:124:in `post'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/tunes/tunes.rb:1229:in `post_sandbox_tester'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/forwardable.rb:230:in `post_sandbox_tester'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/models/sandbox_tester.rb:58:in `create'
	from sample.rb:11:in `<main>'
```

Fastlane env:
```
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 12.2.1                                                         |
| Ruby                        | 2.6.8                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.32.0 (Apple Git-132)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | macOS 12.2.1 (21D62)                                           |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 13.3                                                           |
| Swift Version               | 5.6                                                            |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.205.0 | 🚫 Update available |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.3.0        |
| atomos                          | 0.1.3        |
| CFPropertyList                  | 2.3.6        |
| claide                          | 1.0.3        |
| colored2                        | 3.1.2        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.19.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.8.0        |
| terminal-table                  | 1.8.0        |
| plist                           | 3.6.0        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.8.0        |
| multipart-post                  | 2.0.0        |
| word_wrap                       | 1.0.0        |
| optparse                        | 0.1.1        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| artifactory                     | 3.0.15       |
| babosa                          | 1.0.4        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| excon                           | 0.90.0       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| faraday-multipart               | 1.0.3        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.9.3        |
| unf_ext                         | 0.0.8.1      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| http-cookie                     | 1.0.4        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.2.0        |
| fastimage                       | 2.2.6        |
| gh_inspector                    | 1.1.3        |
| json                            | 2.1.0        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| xcpretty-travis-formatter       | 1.0.1        |
| dotenv                          | 2.7.6        |
| bundler                         | 1.17.2       |
| simctl                          | 1.6.8        |
| jwt                             | 2.3.0        |
| uber                            | 0.1.0        |
| declarative                     | 0.0.20       |
| trailblazer-option              | 0.1.2        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| mini_mime                       | 1.1.2        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.16.1       |
| googleauth                      | 1.1.2        |
| httpclient                      | 2.8.3        |
| rexml                           | 3.1.9.1      |
| webrick                         | 1.4.4        |
| google-apis-core                | 0.4.2        |
| google-apis-playcustomapp_v1    | 0.7.0        |
| google-apis-androidpublisher_v3 | 0.16.0       |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.2.0        |
| google-cloud-core               | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.10.0       |
| google-apis-storage_v1          | 0.11.0       |
| rake                            | 12.3.3       |
| digest-crc                      | 0.6.4        |
| google-cloud-storage            | 1.36.1       |
| emoji_regex                     | 3.2.3        |
| jmespath                        | 1.6.1        |
| aws-partitions                  | 1.568.0      |
| aws-eventstream                 | 1.2.0        |
| aws-sigv4                       | 1.4.0        |
| aws-sdk-core                    | 3.130.0      |
| aws-sdk-kms                     | 1.55.0       |
| aws-sdk-s3                      | 1.113.0      |
| forwardable                     | 1.2.0        |
| logger                          | 1.3.0        |
| date                            | 2.0.0        |
| stringio                        | 0.0.2        |
| ipaddr                          | 1.2.2        |
| openssl                         | 2.1.2        |
| zlib                            | 1.0.0        |
| mutex_m                         | 0.1.0        |
| ostruct                         | 0.1.0        |
| strscan                         | 1.0.0        |
| io-console                      | 0.4.7        |
| fileutils                       | 1.1.0        |
| etc                             | 1.0.1        |
| libxml-ruby                     | 3.2.1        |
| psych                           | 3.1.0        |
</details>


*generated on:* **2022-03-25**
</details>
```
Fastlane version: 2.205.0

Please let me know if i am missing any thing .

<details>
  <pre> Jay </pre>
</details>

**Comment by Farr69** [comment]
Interesting! I will try to figure it out for more.

**Comment by jayantsarda** [comment]
@Farr69 were you able to figure out any solution ?

**Comment by Younnass** [comment]
@jayantsarda I need your help, 

    6^x + 7^x + 8^x + 9^x = date of birth 
Please make this program this is my assignment.

**Comment by Manjula123h** [comment]
I am also facing the same issue @jayantsarda  and i cant use ruby script to automate the process of creating sandbox accounts...How can i use spaceship palyground and can it be integrated in the existing ruby framework

**Comment by jayantsarda** [comment]
@Manjula123h , I was able to still automate the with Ruby without using spaceship playground. The only issue we have is that we still need to do a 2FA every one month.

**Comment by manjula08** [comment]
@jayantsarda ..Are you authenticating using Spaceship::ConnectAPI::Token or Spaceship::ConnectAPI.login(username, password to create sandbox testers..i am able to create sandbox testers using login API but it always prompts for 2FA.

By using Token i am unable to create sandbox testers its throwing the same error you have mentioned 
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/api_client.rb:199:in `handle_response': The URL path is not valid - The resource 'sandboxTesters' does not exist (Spaceship::UnexpectedResponse)
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/api_client.rb:124:in `post'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/tunes/tunes.rb:1229:in `post_sandbox_tester'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/forwardable.rb:230:in `post_sandbox_tester'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/models/sandbox_tester.rb:58:in `create'
	from sample.rb:11:in `<main>'


Please let me know how you have automated it using Token api

**Comment by manjula08** [comment]
The quick guide would be appreciated..

**Comment by Manjula123h** [comment]
@jayantsarda ..

---------------Urgent help needed--------

Are you authenticating using Spaceship::ConnectAPI::Token or Spaceship::ConnectAPI.login(username, password to create sandbox testers..i am able to create sandbox testers using login API but it always prompts for 2FA.

By using Token i am unable to create sandbox testers its throwing the same error you have mentioned
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/api_client.rb:199:in handle_response': The URL path is not valid - The resource 'sandboxTesters' does not exist (Spaceship::UnexpectedResponse) from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/api_client.rb:124:in post'
from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/tunes/tunes.rb:1229:in post_sandbox_tester' from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/forwardable.rb:230:in post_sandbox_tester'
from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.0/spaceship/lib/spaceship/connect_api/models/sandbox_tester.rb:58:in create' from sample.rb:11:in

'
Please let me know how you have automated it using Token api

**Comment by jayantsarda** [comment]
@Manjula123h 

try this:

`Spaceship::Tunes.login("AppStore connect email", "AppStore connect pwd")

    attributes = {}
    attributes["firstName"] = "Some Name"
    attributes["lastName"] = "Some Lastname"
    attributes["email"] = email.to_s
    attributes["password"] = "some_email"
    attributes["confirmPassword"] = "<pwd>"
    attributes["secretQuestion"] = "Company"
    attributes["secretAnswer"] = "some company"
    attributes["birthDate"] = "1980-03-01"
    attributes["appStoreTerritory"] = USA

   Spaceship::ConnectAPI.post_sandbox_tester(attributes: attributes)`

**Comment by Manjula123h** [comment]
@jayantsarda ..

@jayantsarda ..This will prompt for 2FA everytime we make a API call isn't it?..Can we use injecting a session in fastlane api session which lasts for one month..

How could you able to do it without prompting for 2FA

**Comment by jayantsarda** [comment]
@Manjula123h Actually it does not prompt every time... it requires me for 2FA only after few weeks .
Read below post. If you have admin rights on your system, it should create a 2FA session token on your system
https://stackoverflow.com/questions/63508108/two-factor-authentication-with-fastlane

Make sure you are on latest Fastlane version.

**Comment by Manjula123h** [comment]
@jayantsarda ..Thanks for your quick response..

But i am planning to run the tests on Browserstack, So i would need to have session being lasts for long time..I have the developer rights not admin rights

Can we connect offline?

**Comment by manjula08** [comment]
@jayantsarda ..Could you please share the steps you followed?.. Are you using lane context of fastlane?

**Comment by Manjula123h** [comment]
@jayantsarda ..How are you holding the session
